.ci/Jenkinsfile

```diff
@@ -99,22 +96,23 @@ pipeline {
 }
 
 def generateStepForAgent(Map params = [:]){
-  def agent = params.get('agent')
+  def agentName = params.get('agent')
   def featuresPath = params.get('featuresPath')
-  log(level: 'INFO', text: "agent=${agent} featuresPath=${featuresPath}")
+  log(level: 'INFO', text: "agent=${agentName} featuresPath=${featuresPath}")
   return {
     node('linux && immutable') {
       catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
         deleteDir()
         unstash 'source'
         dir("${BASE_DIR}"){
           setupAPMGitEmail(global: true)
-          retry(3){
-            sleep randomNumber(min: 5, max: 10)
-            sh script: '.ci/scripts/install-dependencies.sh', label: "Install dependencies"
-          }
-          withCredentials([string(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7', variable: 'GITHUB_TOKEN')]){
-            sh script: """.ci/scripts/send-pr.sh "${agent}" "${featuresPath}" "${env.DO_SEND_PR}" """, label: "Send Pull Request for apm-agent-${agent}"
+          sh script: """.ci/scripts/prepare-spec-changes.sh "${agentName}" "${featuresPath}" """, label: "Prepare changes for apm-agent-${agentName}"
+          dir(".ci/apm-agent-${agentName}") {
```

Comment:
I'm tempted to say we could pass the agent-repo-name instead the name of the agent